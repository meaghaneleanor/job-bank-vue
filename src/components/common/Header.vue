<template>
  <header class="ontario-main-header">
    <div class="grid-container">
      <div class="grid-x">
        <div class="ontario-main-header__logo ontario-grid-cell small-5 medium-3">
          <a class="ontario-main-header__logo-link" href="https://ontario.ca/page/government-ontario" title="Government of Ontario">
            <img class="ontario-main-header__logo-image" alt="Government of Ontario home page" src="../../../public/logo-ontario@2x.png" />
          </a>
        </div>
        <div class="show-for-medium ontario-main-header__search ontario-grid-cell medium-5 large-6" v-if="search">
          <Search/>
        </div>
        <div class="ontario-main-header__site-nav ontario-grid-cell small-7 medium-4 large-3">
          <SiteMenu/>
        </div>
      </div>
    </div>
  </header>
</template>

<script>
import SiteMenu from './SiteMenu.vue'
import Search from './Search.vue'

export default {
  data: () => ({
    search: true
  }),
  components: {
    SiteMenu,
    Search
  }
}
</script>
