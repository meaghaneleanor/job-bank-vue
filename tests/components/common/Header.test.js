import { shallowMount } from '@vue/test-utils';
import messages from '../../../src/translations';
import Header from '../../../src/components/common/Header.vue';

const createComponent = shallowMount(Header, {
  mocks: {
    $t: () => {},
    $i18n: {
      locale: 'en',
      messages
    }
  }
});

describe('Header.vue', () => {
  describe('snapshot', () => {
    it('should display the expected html', () => {
      let wrapper = createComponent;
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
