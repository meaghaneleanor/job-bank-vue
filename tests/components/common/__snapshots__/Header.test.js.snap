// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue snapshot should display the expected html 1`] = `"<header class=\\"ontario-main-header\\"><div class=\\"grid-container\\"><div class=\\"grid-x\\"><div class=\\"ontario-main-header__logo ontario-grid-cell small-5 medium-3\\"><a href=\\"https://ontario.ca/page/government-ontario\\" title=\\"Government of Ontario\\" class=\\"ontario-main-header__logo-link\\"><img alt=\\"Government of Ontario home page\\" src=\\"/logo-ontario@2x.png\\" class=\\"ontario-main-header__logo-image\\"></a></div> <div class=\\"show-for-medium ontario-main-header__search ontario-grid-cell medium-5 large-6\\"><search-stub></search-stub></div> <div class=\\"ontario-main-header__site-nav ontario-grid-cell small-7 medium-4 large-3\\"><sitemenu-stub></sitemenu-stub></div></div></div></header>"`;
